Documentation/Commands/strings.md

```diff
@@ -46,3 +46,174 @@ This parser currently doesn't accept any options.
 
 * [See here](../templates/strings) for a list of templates bundled with SwiftGen and their documentation.
 * If you want to write custom templates, make sure to check the [stencil context documentation](../SwiftGenKit%20Contexts/Strings.md) to see what data is available after parsing.
+
+## Plurals
+
+SwiftGen supports definitions of plurals in `.stringsdict` files. (Note: only non-nested plural variables are supported for now)
+
+### Supported
+
+<details open>
+
+<summary>Basic Example</summary>
+
+This example should cover the most common use case of plurals that is also supported by most of the translation management services.
+
+```xml
+<key>competition.event.number-of-matches</key>
+<dict>
+    <key>NSStringLocalizedFormatKey</key>
+    <string>%#@Matches@</string>
+    <key>Matches</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>ld</string>
+        <key>zero</key>
+        <string>No matches</string>
+        <key>one</key>
+        <string>%ld match</string>
+        <key>other</key>
+        <string>%ld matches</string>
+    </dict>
+</dict>
+```
+
+</details>
+
+<details>
+
+<summary>Multiple variables with different types in format key</summary>
+
+```xml
+<key>multiple.placeholders-and-variables.string-int</key>
+<dict>
+    <key>NSStringLocalizedFormatKey</key>
+    <string>%#@element@ %#@has_rating@</string>
+    <key>element</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>@</string>
+        <key>other</key>
+        <string>%@</string>
+    </dict>
```

Comment:
Since Foundation is applying the whole list of arguments to the substituted format specifier, it should be legal. With the current implementation, a format key of `%@ %#@has_rating@` will generate a method with only one parameter for the `has_rating` variable. If we want to support these cases as well, we would need to remove all  variables from the format key and parse the remaining placeholders as well. I'd suggest to postpone this to a later PR, because this would also mean that we need to rethink how we match the sorting of these two different sets of placeholders to the sorting of the parameters.

Reply:
As I said in the other comment, I'm not sure how Foundation could even determine the proper plurality to use if the placeholder is not a number. Sure here there's only `other` as an option for this variable, but in principle, given that this `%@` is not representing a number, what would the logic be in general to determine if it should use `zero`, `one`, `many`, etc? It can't really test if `"myString" > 1` to choose ;)

That's probably why [the documentation explicitly says](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/StringsdictFileFormat/StringsdictFileFormat.html#//apple_ref/doc/uid/10000171i-CH16-SW4) that this has to be a number:

> `NSStringFormatValueTypeKey`
> A string format specifier **for a number**, as in `%d` for an integer.

_(emphasis mine)_

And probably explains why you have `(null)` at runtime in this example